### PR TITLE
feat(stark-ui): add support for displaying the "items per page" dropdown also in Pagination on "compact" mode

### PR DESCRIPTION
**Pull request by christophercr**
ISSUES CLOSED: #1248

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1248

## What is the new behavior?
It is now possible to display the `items per page` dropdown in the pagination also on "compact" mode. Since the Table component displays the Pagination component on compact "mode", this change allows the developers to display the "items per page" dropdown in the table by setting `itemsPerPageIsPresent: true` in the pagination config. 

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

**Comment by christophercr** [comment]
Thinking a bit more about the actual issue, it turns out the solution could be much simpler without break anything ;)

This PR contains the change I propose, only that, so be aware that I didn't have time to implement unit tests, translations, etc...

If you guys agree on this change, could you maybe complete this PR with the changes needed?

**Comment by nicanac** [comment]
> Thinking a bit more about the actual issue, it turns out the solution could be much simpler without breaking anything ;)
> 
> This PR contains the change I propose, only that, so be aware that I didn't have time to implement unit tests, translations, etc...
> 
> @nicanac @SuperITMan If you guys agree on this change, could you maybe complete this PR with the changes needed?

sure, I'm on holiday till the 9 of December. I'll do it when I'll be back.

**Comment by nicanac** [comment]
Hi @christophercr , 
I come back on this issue, I've small problem while trying to make it work on the showcase. Is it working on your side ?
Thanks

**Comment by christophercr** [comment]
@nicanac what "issue" exactly? As far as I remember, it was working on my side

**Comment by nicanac** [comment]
> @nicanac what "issue" exactly? As far as I remember, it was working on my side

Actually, the drop down is not displayed

**Comment by christophercr** [comment]
It is displayed:

Screenshot 2019-12-16 at 10 22 21

Please note that it's the second table of the demo page

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts [hunk 1]
```diff
@@ -169,12 +169,18 @@ export class StarkPaginationComponent extends MatPaginator implements OnInit, On
 			return { totalItems: 0 };
 		}
 
+		let defaultItemsPerPageIsPresent: StarkPaginationConfig["pageInputIsPresent"];
```
[the remark this answers is not in the document]
Comment (christophercr) [reply packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: Oops... indeed. @nicanac can you adapt this in the changes you are doing to this PR?

File: packages/stark-ui/src/modules/pagination/components/pagination.component.html [hunk 2]
```diff
@@ -100,4 +100,15 @@
 			</button>
 		</li>
 	</ul>
+
+	<div class="pagination-items-per-page" *ngIf="paginationConfig.itemsPerPageIsPresent !== false">
```
[the remark this answers is not in the document]
Comment (christophercr) [reply packages/stark-ui/src/modules/pagination/components/pagination.component.html]: True... @nicanac can you also adapt this in the changes you are doing to this PR?

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicanac) [reply packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: Hi @SuperITMan, @christophercr 
It's done.

File: packages/stark-ui/src/modules/pagination/components/pagination.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicanac) [reply packages/stark-ui/src/modules/pagination/components/pagination.component.html]: Hi @SuperITMan, @christophercr 
It's done.

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts [hunk 3]
```diff
@@ -130,7 +130,7 @@ export class StarkPaginationComponent extends MatPaginator implements OnInit, On
 	}
 
 	/**
-	 * Component lifecycle hook
+	 * CodefaultItemsPerPageIsPresentmponent lifecycle hook
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: Could you please switch this back to `Component` ? 😊

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts [hunk 4]
```diff
@@ -165,16 +165,22 @@ export class StarkPaginationComponent extends MatPaginator implements OnInit, On
 	public normalizePaginationConfig(config: StarkPaginationConfig): StarkPaginationConfig {
 		if (!config) {
 			// initialize paginationConfig to prevent errors in other functions depending on this config
-			this.logger.warn(componentName + ": No configuration defined in pagination component. TotalItems set to 0 by default");
+			this.logger.warn(componentName + ": No configuration defined in pagination component. TotalIt  ems set to 0 by default");
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: Could you please put back this previous code ?
`this.logger.warn(componentName + ": No configuration defined in pagination component. TotalItems set to 0 by default");`

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<stark-table
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.html]: Could you please add an example of this file in the "showcase/src/assets/examples/table/..." ?

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
+import { Component, Inject } from "@angular/core";
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: Same here, could you please add an example of this file in the "showcase/src/assets/examples/table/..." ?

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+	[data]="data"
+	[columnProperties]="columns"
+	[filter]="filter"
+	[orderProperties]="order"
+	[paginationConfig]="pagination"
+	[tableRowActions]="tableRowActions"
+	showRowIndex
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.html]: Is the `showRowIndex` necessary here ?
I have the feeling it's useless in this case 😕

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+				icon: "delete",
+				actionCall: ($event: Event, data: object): void => this.logger.debug("DELETE", $event, data),
+				isEnabled: false
+			}
+		]
+	};
+
+	public constructor(@Inject(STARK_LOGGING_SERVICE) private logger: StarkLoggingService) {}
+
+	public handleRowClicked(data: object): void {
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: This should be present only in the regular table or in all table examples.
I suggest to remove this from the table with items-per-page selector 😊

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.html [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+	multiSort
+	showRowsCounter
+	(rowClicked)="handleRowClicked($event)"
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.html]: Same here about the `handleRowClicked`

File: showcase/src/assets/translations/en.json [hunk 10]
```diff
@@ -330,6 +330,7 @@
         },
         "REGULAR": "Regular Table",
         "WITH_SELECTION": "Table with selection",
+        "WITH_ITEMS_PER_PAGE_SELECTOR": "Table with custom number of items per page",
```
Comment (SuperITMan) [inline showcase/src/assets/translations/en.json]: I suggest to change this for:
"Table with the selector of items per page to display"

File: showcase/src/assets/translations/fr.json [hunk 11]
```diff
@@ -330,6 +330,7 @@
         },
         "REGULAR": "Table régulière",
         "WITH_SELECTION": "Table avec sélection",
+        "WITH_ITEMS_PER_PAGE_SELECTOR": "Table avec la sélection du nombre d'éléments à afficher",
```
Comment (SuperITMan) [inline showcase/src/assets/translations/fr.json]: I suggest to change this for:
"Table avec le sélecteur du nombre d'éléments par page à afficher"

File: showcase/src/assets/translations/nl.json [hunk 12]
```diff
@@ -330,6 +330,7 @@
         },
         "REGULAR": "Normale tabel",
         "WITH_SELECTION": "Tabel met selectie",
+        "WITH_ITEMS_PER_PAGE_SELECTOR": "Tabel met aangepaste items per page",
```
Comment (SuperITMan) [inline showcase/src/assets/translations/nl.json]: I suggest to change this for:
"Tabel met de selector van items per pagina om weer te geven"

File: showcase/src/app/demo-ui/pages/table/demo-table-page.component.html [hunk 13]
```diff
@@ -10,6 +10,15 @@ <h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
 		<showcase-table-regular></showcase-table-regular>
 	</example-viewer>
 
+	<example-viewer
+		id="items-per-page-selection"
+		exampleTitle="SHOWCASE.DEMO.TABLE.WITH_ITEMS_PER_PAGE_SELECTOR"
+		filesPath="fix-me"
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/pages/table/demo-table-page.component.html]: Once the examples are added, don't forget to change this path 😊

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts [hunk 14]
```diff
@@ -130,15 +130,15 @@ export class StarkPaginationComponent extends MatPaginator implements OnInit, On
...
+	 * CodefaultItemsPerPageIsPresentComponent lifecycle hook
 	 */
 	public ngOnChanges(changesObj: SimpleChanges): void {
 		if (changesObj["paginationConfig"]) {
 			// Set local variable to prevent shadow changes
 			const paginationConfigOriginalChange: StarkPaginationConfig = { ...this.paginationConfig };
 			this.paginationConfig = this.normalizePaginationConfig(this.paginationConfig);
 			this.logger.debug(componentName + ": paginationConfig changed...", this.paginationConfig);
-
+			this.logger.warn(componentName + ": No configuration defined in pagination component. TotalItems set to 0 by default");
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: Actually, after rereading the code, I see that this line can be removed. It's a copy of line 168.
That means this message will be displayed twice.

Could you please remove this ? 😊

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts [hunk 15]
```diff
@@ -130,15 +130,15 @@ export class StarkPaginationComponent extends MatPaginator implements OnInit, On
 	}
 
 	/**
-	 * Component lifecycle hook
+	 * CodefaultItemsPerPageIsPresentComponent lifecycle hook
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: This is incorrect, it should stay "Component lifecycle hook". Could you please adapt this ?

File: showcase/src/app/demo-ui/pages/table/demo-table-page.component.html [hunk 16]
```diff
@@ -10,6 +10,15 @@ <h1 translate>SHOWCASE.DEMO.SHARED.EXAMPLE_VIEWER_LIST</h1>
 		<showcase-table-regular></showcase-table-regular>
 	</example-viewer>
 
+	<example-viewer
+		id="items-per-page-selection"
+		exampleTitle="SHOWCASE.DEMO.TABLE.WITH_ITEMS_PER_PAGE_SELECTOR"
+		filesPath="table/with-items-per-page-selector/table"
+		[extensions]="['HTML', 'TS', 'SCSS']"
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/pages/table/demo-table-page.component.html]: Since we have only examples for `'TS'` and `'HTML'`, we can remove `'SCSS'` 😊

File: showcase/src/assets/examples/table/with-items-per-page-selector/table.html [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+	[data]="data"
+	[columnProperties]="columns"
+	[filter]="filter"
+	[orderProperties]="order"
+	[paginationConfig]="pagination"
+	[tableRowActions]="tableRowActions"
+	multiSort
+	showRowsCounter
+>
+	<header><h1 class="mb0" translate>SHOWCASE.DEMO.TABLE.WITH_ITEMS_PER_PAGE_SELECTOR</h1></header>
```
Comment (SuperITMan) [inline showcase/src/assets/examples/table/with-items-per-page-selector/table.html]: Could you please replace the translation key by "Table with the selector of items per page to display" ?
As we have in the other examples 😊

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts [hunk 18]
```diff
@@ -0,0 +1,90 @@
+import { Component, Inject } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkPaginationConfig, StarkTableColumnProperties, StarkTableFilter, StarkTableRowActions } from "@nationalbankbelgium/stark-ui";
+
+const DUMMY_DATA: object[] = [
+	{
+		id: 1,
+		title: { label: "first title (value: 1)", value: 1 },
+		description: "first one",
+		extra: "number one has some extra information"
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: Is the `extra` property useful here ? Or is it linked to the implementation of collapsable rows in the table ?
If so, could you please remove it since it's not related to the items-per-page ?

File: showcase/src/assets/examples/table/with-items-per-page-selector/table.ts [hunk 19]
```diff
@@ -0,0 +1,90 @@
+import { Component, Inject } from "@angular/core";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkPaginationConfig, StarkTableColumnProperties, StarkTableFilter, StarkTableRowActions } from "@nationalbankbelgium/stark-ui";
+
+const DUMMY_DATA: object[] = [
```
Comment (SuperITMan) [inline showcase/src/assets/examples/table/with-items-per-page-selector/table.ts]: Could you please reduce this object as we did in another example: https://github.com/NationalBankBelgium/stark/blob/52ae6e0380e24bd3678f0df15edd53746c6ac79c/showcase/src/assets/examples/table/with-custom-styling/table.ts#L4 

```typescript
const DUMMY_DATA: object[] = [
	{ id: 1, title: { label: "first title (value: 1)", value: 1 }, description: "number one" },
	/* ... */
	{ id: 44, title: { label: "twelfth title (value: 12)", value: 12 }, description: "the twelfth description" }
];
```

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+		{ name: "id", label: "Id" },
+		{
+			name: "title",
+			label: "SHOWCASE.DEMO.TABLE.LABELS.TITLE",
+			cellFormatter: (value: { label: string }): string => "~" + value.label,
+			compareFn: (n1: { value: number }, n2: { value: number }): number => n1.value - n2.value
+		},
+		{ name: "description", label: "SHOWCASE.DEMO.TABLE.LABELS.DESCRIPTION" },
+		{
+			name: "extra",
```
Comment (SuperITMan) [inline showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: Same question about the column `extra`. Can it be removed if it's not related to items-per-page ? 😊

File: showcase/src/assets/examples/table/with-items-per-page-selector/table.html (hunk 17 above)
Comment (nicanac) [reply showcase/src/assets/examples/table/with-items-per-page-selector/table.html]: Yes, I was also wondering, isn't there any way to use the translation in the example too ?

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts (hunk 18 above)
Comment (nicanac) [reply showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: No, Actually it's coming from the table-regular example.

**Comment by nicanac** [comment]
Hey @SuperITMan , I've reviewed the changes.
Could you have a look at it ?
Thanks

File: showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts (hunk 18 above)
Comment (SuperITMan) [reply showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: I looked at the code of regular table and it is there to show the feature "show/hide column".
We should keep it for the regular table 😊

Comment (nicanac) [reply showcase/src/app/demo-ui/components/table-with-items-per-page-selector/table-with-items-per-page-selector.component.ts]: so let's keep it :-)

File: showcase/src/assets/examples/table/with-items-per-page-selector/table.html (hunk 17 above)
Comment (SuperITMan) [reply showcase/src/assets/examples/table/with-items-per-page-selector/table.html]: No, there is no way because the `example-viewer` component displays the raw file.

File: packages/stark-ui/src/modules/pagination/components/pagination.component.ts [hunk 21]
```diff
@@ -138,7 +138,6 @@ export class StarkPaginationComponent extends MatPaginator implements OnInit, On
 			const paginationConfigOriginalChange: StarkPaginationConfig = { ...this.paginationConfig };
 			this.paginationConfig = this.normalizePaginationConfig(this.paginationConfig);
 			this.logger.debug(componentName + ": paginationConfig changed...", this.paginationConfig);
-
```
Comment (SuperITMan) [inline packages/stark-ui/src/modules/pagination/components/pagination.component.ts]: Could you please revert this space to keep the code easy to read ? 😊

File: showcase/package-lock.json [hunk 22]
```diff
@@ -1524,8 +1524,7 @@
       "integrity": "sha512-flCoA7F4wRkymaP6zwAWPrgqDLx4eQM5Ep/pLtmMyVp/FQz0eevwyNfa4lEaYSFHzcs2bKVJr8O35JrioQL/Yg=="
     },
     "@nationalbankbelgium/stark-build": {
-      "version": "file:../dist/packages-dist/stark-build/nationalbankbelgium-stark-build-10.0.0-rc.2-0c7801b2.tgz",
-      "integrity": "sha512-GcmIdKEVH95WNRt9n/jRBQez7HpGIFF5aXPnMt5k+nkXJObQXc1VQyUZH7+deCOFwhgij7UwBAHpi7hq/ZpdAw==",
+      "version": "file:../dist/packages-dist/stark-build/nationalbankbelgium-stark-build-10.0.0-rc.2-c99475f8.tgz","integrity": "sha512-CEY55bmM9ECWfVUFgP/CZ2y/ab4TzfHQD+WQC1H2b2mE8sSm1yyOkXNbl71wM/ql6z0czfpFTK45SXvOHUcNEw==",
```
Comment (SuperITMan) [inline showcase/package-lock.json]: Could you please revert the changes in this file since they are not necessary ? 😊

File: showcase/package.json [hunk 23]
```diff
@@ -125,9 +125,9 @@
     "@angular/platform-server": "~7.2.2",
     "@angular/router": "~7.2.2",
     "@nationalbankbelgium/code-style": "^1.0.1",
-    "@nationalbankbelgium/stark-core": "file:../dist/packages-dist/stark-core/nationalbankbelgium-stark-core-10.0.0-rc.2-0c7801b2.tgz",
```
Comment (SuperITMan) [inline showcase/package.json]: Could you please revert the changes in this file since they are not necessary ? 😊

File: showcase/src/assets/examples/table/with-items-per-page-selector/table.html (hunk 17 above)
Comment (nicanac) [reply showcase/src/assets/examples/table/with-items-per-page-selector/table.html]: ok, I see, crap ...
